### ISSUE TITLE
builtins.break doesn't break

### ISSUE THREAD

**Issue by yajo** [NixOS/nix#6649]
**Describe the bug**

I add `builtins.break` to some code, and it doesn't break.

**Steps To Reproduce**

1.  Add this simple `flake.nix`:

    ```nix
    {
      inputs.nixpkgs.url = "nixpkgs/f1c9c23aad972787f00f175651e4cb0d7c7fd5ea";
      outputs = {
        self,
        nixpkgs,
      }: let
        pkgs = builtins.break nixpkgs.legacyPackages.x86_64-linux;
      in {
        devShells.x86_64-linux.default = pkgs.mkShell {
          buildInputs = [pkgs.hello];
        };
      };
    }
    ```

2.  Run `nix develop --debugger`

https://user-images.githubusercontent.com/973709/173176380-870c1b4c-2ebd-41e1-9980-f57428d38d83.mp4

**Expected behavior**

Nix repl should stop at the break point.

**`nix-env --version` output**
nix-env (Nix) 2.9.0

**Additional context**

The screencast seems to display bug #6594 but the main problem is that it doesn't stop at the desired break point.

**Comment by thufschmitt** [comment]
cc @bburdette

**Comment by bburdette** [comment]
I've only had a chance to look at this briefly, but I suspect that what's happening here is an exception is thrown that isn't an EvalException, and so it isn't caught by the debugger.  But you should at least get the error message from it, and that doesn't happen either.

**Comment by 9999years** [comment]
This is fixed in #9917.
